estonian_e_invoice/entities/account.py

```diff
@@ -13,20 +18,27 @@ class AccountInfo:
         bank_name: The name of the bank.
     """
 
+    tag = "AccountInfo"
+    validation_schema = ACCOUNT_INFO_SCHEMA
+
     def __init__(
         self,
         account_number: str,
         iban: Optional[str] = None,
         bic: Optional[str] = None,
         bank_name: Optional[str] = None,
     ) -> None:
-        self.account_number = account_number
-        self.iban = iban
-        self.bic = bic
-        self.bank_name = bank_name
+        self.elements = self.validate(
```

Comment:
Not a requirement, but could have some overridable `attribute_mapping` class attribute on Node, that would hold the mapping between `**kwargs` and XML Attreibutes, like

```
attribute_mapping = {
    "account_number": "AccountNumber",
    ...
}
```

then could have also on Node class

```
def __init__(self, **kwargs):
    self.elements = self.validate({self.attribute_mapping[k]: v for k,v in kwargs})
```

which would reduce amount of code in subclasses greatly - most will only need to also override/extend `attribute_mapping`

Reply:
Yeah totally make sense. However, would it be useful from the users' perspective?

It might be hard to understand what to provide for the XML element class.

Reply:
Yeah, I guess there is no good way to keep the logic generic but also keep the arguments annotated and explicitly declared.